apps/server/src/utils/ElasticSearchClient.test.ts

```diff
@@ -0,0 +1,202 @@
+import { elasticSearchInstitutionData } from '../test/testData/institution'
+import { ElasticSearchMock, getFavoriteInstitutions, getInstitution, initialize, reIndexElasticSearch, search } from '../utils/ElasticSearchClient'
+
+describe('initialize', () => {
+  describe('elastic search already indexed', () => {
+    let indexCreated: boolean = false
+
+    beforeAll(() => {
+      ElasticSearchMock.clearAll()
+      ElasticSearchMock.add({
+        method: 'HEAD',
+        path: '/institutions'
+      }, () => {
+        return ''
+      })
+
+      ElasticSearchMock.add({
+        method: 'PUT',
+        path: '/institutions'
+      }, () => {
+        indexCreated = true
+        return ''
+      })
+    })
+
+    it('does not reindex institutions', async () => {
+      await initialize()
+      expect(indexCreated).toBeFalsy()
+    })
+  })
+
+  describe('elastic search not indexed', () => {
+    let indexCreated: boolean
+
+    beforeAll(() => {
+      ElasticSearchMock.clearAll()
+      ElasticSearchMock.add({
+        method: 'PUT',
+        path: '/institutions'
+      }, () => {
+        indexCreated = true
+        return ''
+      })
+
+      ElasticSearchMock.add({
+        method: 'PUT',
+        path: '/institutions/_doc/*'
+      }, () => {
+        return ''
+      })
+    })
+
+    it('triggers the reIndexElasticSearch method which makes call ES create index endpoint', async () => {
+      await initialize()
+      expect(indexCreated).toBeTruthy()
+    })
+  })
+})
+
+describe('reIndexElasticSearch', () => {
+  let indexCreated: boolean
+  let institutionsIndexedCount: number
+
+  beforeAll(() => {
+    ElasticSearchMock.clearAll()
+    indexCreated = false
+    institutionsIndexedCount = 0
+
+    ElasticSearchMock.add({
+      method: 'PUT',
+      path: '/institutions'
+    }, () => {
+      indexCreated = true
+      return ''
+    })
+
+    ElasticSearchMock.add({
+      method: 'PUT',
+      path: '/institutions/_doc/*'
+    }, () => {
+      institutionsIndexedCount += 1
+      return ''
+    })
+  })
+
+  it('makes call to create index and makes call to index more than 4 institutions', async () => {
+    await reIndexElasticSearch()
+    expect(indexCreated).toBeTruthy()
+    expect(institutionsIndexedCount).toBeGreaterThan(4)
+  })
+})
+
+describe('search', () => {
+  beforeEach(() => {
+    ElasticSearchMock.clearAll()
+    ElasticSearchMock.add({
```

Comment:
I think it makes more sense to put the mock associated with each test in the test instead of in this beforeall